mmdet3d/models/backbones/dla.py

```diff
@@ -0,0 +1,313 @@
+                    kernel_size=1,
+                    stride=1,
+                    bias=False),
+                dla_build_norm_layer(norm_cfg, out_channels)[1])
+
+    def forward(self, x, residual=None, children=None):
+        children = [] if children is None else children
+        bottom = self.downsample(x) if self.downsample else x
+        residual = self.project(bottom) if self.project else bottom
+        if self.level_root:
+            children.append(bottom)
+        x1 = self.tree1(x, residual)
+        if self.levels == 1:
+            x2 = self.tree2(x1)
+            x = self.root(x2, x1, *children)
+        else:
+            children.append(x1)
+            x = self.tree2(x1, children=children)
+        return x
+
+
+@BACKBONES.register_module()
+class DLANet(nn.Module):
+    """DLA backbone.
+
+    Args:
+        depth (int): Depth of dla. Default: 34.
+        in_channels (int): Number of input image channels. Default: 3.
+        norm_cfg (dict): Dictionary to construct and config norm layer.
+        residual_root (bool): whether to use residual in root layer
+    """
+    arch_settings = {
+        34: (BasicBlock, (1, 1, 1, 2, 2, 1), (16, 32, 64, 128, 256, 512)),
+    }
+
+    def __init__(self,
+                 depth=34,
+                 in_channels=3,
+                 norm_cfg=None,
+                 residual_root=False):
+        super(DLANet, self).__init__()
+        if depth not in self.arch_settings:
+            raise KeyError(f'invalida depth {depth} for DLA')
+        block, levels, channels = self.arch_settings[depth]
+        self.channels = channels
+        # self.num_classes = num_classes
+        self.base_layer = nn.Sequential(
+            nn.Conv2d(
+                in_channels,
+                channels[0],
+                kernel_size=7,
+                stride=1,
+                padding=3,
+                bias=False),
+            dla_build_norm_layer(norm_cfg, channels[0])[1],
+            nn.ReLU(inplace=True))
+        self.level0 = self._make_conv_level(channels[0], channels[0],
+                                            levels[0], norm_cfg)
+        self.level1 = self._make_conv_level(
+            channels[0], channels[1], levels[1], norm_cfg, stride=2)
+        self.level2 = Tree(
+            levels[2],
+            block,
+            channels[1],
+            channels[2],
+            norm_cfg,
+            2,
+            level_root=False,
+            root_residual=residual_root)
+        self.level3 = Tree(
+            levels[3],
+            block,
+            channels[2],
+            channels[3],
+            norm_cfg,
+            2,
+            level_root=True,
+            root_residual=residual_root)
+        self.level4 = Tree(
+            levels[4],
+            block,
+            channels[3],
+            channels[4],
+            norm_cfg,
+            2,
+            level_root=True,
+            root_residual=residual_root)
+        self.level5 = Tree(
```

Comment:
Can we simplify line227-262 using `for` iteration? It seems to me that the indexing is similar. See [ResNet](https://github.com/open-mmlab/mmdetection/blob/62a1cd3fd6091fc4efa83627631ab4b99a8d005c/mmdet/models/backbones/resnet.py#L459).